drivers/imu/adis.c

```diff
@@ -2552,3 +2558,88 @@ int adis_get_sync_clk_freq(struct adis_dev *adis, uint32_t *clk_freq)
 
 	return 0;
 }
+
+/**
+ * @brief Read adis device gyroscope scale in fractional form.
+ * @param adis       - The adis device.
+ * @param gyro_scale - The gyroscope scale.
+ * @return 0 in case of success, error code otherwise.
+ */
+int adis_get_gyro_scale(struct adis_dev *adis,
+			struct adis_scale_fractional *gyro_scale)
+{
+	if (!adis || !gyro_scale)
+		return -EINVAL;
+
+	*gyro_scale = adis->gyro_scale;
```

Comment:
can't we obtain this from hardware instead of caching it? Same question applies for all getter functions added in this patch.

Reply:
these values are defined in adis1650x.c and in adis1657x.c. and are specific to each chip version. they are not available in the hardware register map (if that is what you are asking)